src/Infra/MySQL/MySQLSettingRepository.ts

```diff
@@ -27,4 +29,19 @@ export class MySQLSettingRepository extends Repository<Setting> implements Setti
       )
       .getMany()
   }
+  async createOrReplace(dto: CreateOrReplaceSettingDto): 
+  Promise<CreateOrReplaceSettingStatus> {
+    const { user, props } = dto
+
+    const existing = await this.findOneByNameAndUserUuid(props.name, user.uuid)
+
+    if (existing === undefined) {
+      await this.save(Setting.create(props, user))
+      return 'created'
```

Comment:
would be good to use consts for this if we need to indicate this status .. as mentioned in previous comments

Reply:
In TS this is not necessary -- `CreateOrReplaceSettingStatus` is a union type of [literal types](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#literal-types) ('created' and 'replaced'). There is autocompletion and spelling mistakes or modifications will cause type errors. So these literals act like lispy keywords or enums.

It does seem though that auto-rename is not supported for them (yet?), so that would be the only advantage I see of putting them in variables, although I don't think that's necessary in this case.

If this is not in style however I can add a level of indirection and turn it into an enum. If you think that's the way to go (or have something else in mind), let me know.

Reply:
all good by me :)